src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,314 @@
+                "errors": [],
+                "extra": {},
+                "ingest_path": [],
+                "level": "info",
+                "location": transaction_name,
+                "logger": "",
+                "measurements": segment_measurements,
+                "metadata": {
+                    "location": transaction_name,
+                    "title": transaction_name,
+                },
+                "nodestore_insert": segment_span.get("precise.finish_ts"),
+                "received": segment_span.get("precise.finish_ts"),
+                "request": {
+                    # missing from indexed spans dataset
+                },
+                "sdk": {
+                    "name": segment_sentry_tags.get("sdk.name"),
+                    "version": segment_sentry_tags.get("sdk.version"),
+                    # other props missing from indexed spans dataset
+                },
+                "span_grouping_config": {
+                    # missing from indexed spans dataset
+                },
+                "spans": _span_data_to_event_spans(spans_data),
+                "start_timestamp": segment_span.get("precise.start_ts"),
+                "timestamp": segment_span.get("precise.finish_ts"),
+                "title": transaction_name,
+                "transaction": transaction_name,
+                "transaction_info": {
+                    # missing from indexed spans dataset
+                },
+                "type": "transaction",
+                "user": {
+                    # missing from indexed spans dataset
+                },
+                "version": 5,
+            },
+            status=200,
+        )
+
+
+def _query_all_spans_in_transaction(
+    organization, project, transaction_id, given_start_str, given_end_str
+):
+    # If we were given a time range by the client, use that time range.
+    try:
+        given_start = datetime.fromtimestamp(float(given_start_str))
+        given_end = datetime.fromtimestamp(float(given_end_str))
+    except TypeError:
+        given_start = None
+        given_end = None
+
+    if given_start and given_end:
+        return _query_spans_in_range(organization, project, transaction_id, given_start, given_end)
+
+    # We don't know when these spans occurred and it takes too long to
+    # search without a fairly narrow time range. Search progressively
+    # farther and farther back until we find the spans.
+    now = datetime.now()
+    all_spans_data = []
+
+    for days_back in range(0, 90):
+        start = now - timedelta(days=days_back + 1)
+        end = now - timedelta(days=days_back)
+        spans_data = _query_spans_in_range(organization, project, transaction_id, start, end)
+        if spans_data:
+            all_spans_data.extend(spans_data)
+        # Assume we've found all the spans if we get an empty result back after already finding some
+        if not spans_data and all_spans_data:
+            break
+
+    return all_spans_data
```

Comment:
i'm kind of curious how practical this is, if i'm reading correctly we'll make up to 90 queries trying to look for these spans? have we tried putting an index on transaction_id in this dataset?

Reply:
Yeah totally, this wasn't going to survive into prod. It turns out we always have timestamps available from the frontend, so I'm going to make them mandatory and kill this progressive querying.

Reply:
> have we tried putting an index on transaction_id in this dataset?

That's the most likely next step after we get performance numbers testing this endpoint on prod data (we'd do that before exposing it to real users).